Havit.Blazor.Components.Web.Bootstrap/wwwroot/HxSidebarItem.js

```diff
@@ -0,0 +1,75 @@
+﻿var dropdowns = new Map();
+
+export function initializePopper(navId, dropdownId) {
+	console.log('initialize popper: ' + dropdownId);
+
+	const nav = document.querySelector(`#${navId}`);
+	const dropdown = document.querySelector(`#${dropdownId}`);
+
+	// Create popper instance
+	const popperInstance = Popper.createPopper(nav, dropdown, {
+		placement: 'right-start',
+		modifiers: [
+			{ name: 'eventListeners', enabled: false }
+		]
+	});
+
+	dropdowns.set(dropdownId, {
+		popperInstance:popperInstance,
+		navElement: nav,
+		show: show,
+		hide: hide
+	});
+
+	nav.addEventListener('mouseenter', show);
+	nav.addEventListener('focus', show);
+
+	nav.addEventListener('mouseleave', hide);
+	nav.addEventListener('blur', hide);
+
+	function show() {
+		// Make the dropdown visible
+		dropdown.setAttribute('popper-show', '');
+
+		// Enable the event listeners
+		popperInstance.setOptions((options) => ({
+			...options,
+			modifiers: [
+				...options.modifiers,
+				{ name: 'eventListeners', enabled: true }
+			]
+		}));
+
+		// Update its position
+		popperInstance.update();
+	}
+
+	function hide() {
+		// Hide the dropdown
+		dropdown.removeAttribute('popper-show');
+
+		// Disable the event listeners
+		popperInstance.setOptions((options) => ({
+			...options,
+			modifiers: [
+				...options.modifiers,
+				{ name: 'eventListeners', enabled: false }
+			]
+		}));
+	}
+}
+
+export function destroyPopper(navId, dropdownId) {
+	// Destroy the popper instance
+	const dropdown = dropdowns.get(dropdownId);
+	dropdown.popperInstance.destroy();
+
+	// Remove the event listeners
+	dropdown.navElement.removeEventListener('mouseenter', dropdown.show);
+	dropdown.navElement.removeEventListener('focus', dropdown.show);
+
+	dropdown.navElement.removeEventListener('mouseleave', dropdown.hide);
+	dropdown.navElement.removeEventListener('blur', dropdown.hide);
+
+	dropdowns.delete(dropdownId);
```

Comment:
...all the code has to be guarded with `if (dropdown)`, otherwise we will hit usual trouble where the DOM is manipulated earlier that the destroy/dispose logic runs.

Reply:
Actually when this code is called the nav element is not present in the DOM. That is why I capture a reference to the element in the `dropdowns` variable and use this reference to cleanup the event listeners. Since I capture this reference to the element I believe I don't have to guard it with an `if (dropdown)` clause.

On the other hand I can also leave it up to the browser -> modern browsers clean up event listeners when the element they are attached to is removed from DOM (when no other references to the element exist).